python/ray/dag/tests/experimental/test_accelerated_dag.py

```diff
@@ -328,6 +328,217 @@ def test_multi_args_and_kwargs(ray_start_regular):
     compiled_dag.teardown()
 
 
+def test_dag_capacity(ray_start_regular):
+    """
+    Test the capacity of a CompiledDAG and exceptions when the capacity is exceeded.
+
+    This test checks a series of linear and branched DAGs. For each DAG, its structure
+    and capacity of internal components are illustrated with a simple ASCII diagram.
+    A number in a parenthesis indicates the Nth intermediate object, an 'x' indicates
+    the actor or channel is blocked to hold any objects.
+    """
+    # Linear: basic case
+    a = Actor.remote(0)
+    with InputNode() as i:
+        dag = a.inc.bind(i)
+    compiled_dag = dag.experimental_compile(info_level_when_execution_may_block="ERROR")
+    # driver -(3)-> a.inc (2) -(1)-> driver
+    assert compiled_dag.capacity == 3
+    refs = []
+    for i in range(3):
+        ref = compiled_dag.execute(1)
+        refs.append(ref)
+    # No blocking when capacity is not exceeded.
+    result = ray.get(ref)
+    assert result == 3
+    # Throws an error when capacity is exceeded.
+    refs = []
+    for i in range(3):
+        ref = compiled_dag.execute(1)
+        refs.append(ref)
+    with pytest.raises(
+        ValueError,
+        match=(
+            "Execution may be blocked due to DAG is at its capacity \(3\). "
+            "Please call ray.get\(\) on previous CompiledDAGRefs to retrieve"
+            " and release them from the DAG."
+        ),
+    ):
+        compiled_dag.execute(1)
+    compiled_dag.teardown()
+
+    # Linear: multiple actors
```

Comment:
Please separate into different tests so that it is easier to debug if one test fails.